src/picotrav.cpp

```diff
@@ -963,6 +1092,42 @@ run_picotrav(int argc, char** argv)
       std::cerr << "Number of outputs in '" << file_0 << "' and '" << file_1 << "' do not match!\n";
       return -1;
     }
+
+    if (match_io_names) {
+      // name-based matching of inputs
+      for (auto& [id_0, pos_0] : net_0.inputs_w_order) {
+        const std::string& name = net_0.nodes[id_0].name;
+        const auto name_map_it  = net_1.name_map.find(name);
+        if (name_map_it != net_1.name_map.end()) {
+          const auto order_it = net_1.inputs_w_order.find(name_map_it->second);
+          if (order_it != net_1.inputs_w_order.end()) {
+            order_it->second = pos_0;
+            continue; // success, continue with the next input
+          }
+        }
+
+        std::cerr << "Input '" << name << "' is present in '" << file_0 << "' but not in '"
+                  << file_1 << "'\n";
+        return -1;
+      }
+
+      // name-based matching of outputs
+      for (unsigned i = 0; i < net_0.outputs_in_order.size(); ++i) {
+        const std::string& name = net_0.nodes[net_0.outputs_in_order[i]].name;
+        const auto name_map_it  = net_1.name_map.find(name);
+        if (name_map_it != net_1.name_map.end()) {
+          const node_id_t id_1 = name_map_it->second;
+          if (net_1.nodes[id_1].is_output) {
+            net_1.outputs_in_order[i] = id_1;
+            continue; // success, continue with the next input
+          }
+        }
+
+        std::cerr << "Output '" << name << "' is present in '" << file_0 << "' but not in '"
+                  << file_1 << "'\n";
+        return -1;
+      }
```

Comment:
This may be moved into a separate function - it is arguably "too low level" for the *main* function.